File: Search Algorithms/insertion_sort.rb
def insertion_sort(array)
  (1...array.length).each do |i|
    j = i

    while (j > 0 && array[j] < array[j - 1])
      array[j], array[j - 1] = array[j - 1], array[j]
      j -= 1
    end
  end
  return array
end

File: Javascript/Sorting Algorithms/Search Algorithms/insertion_sort.js
function insertionSort(array) {
	for (let i = 1; i < array.length; i++) {
		let j = i;

		while (j > 0 && array[j] < array[j - 1]) {
			[ array[j], array[j - 1] ] = [ array[j - 1], array[j] ];
			j--;
		}
	}
	return array;
}

// Time Complexity: O(n^2)
// Space Complexity: O(1)
